components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -955,67 +976,67 @@ void RRTMGPRadiation::run_impl (const double dt) {
+      auto sfc_alb_dif_k     = m_buffer.sfc_alb_dif_k;
+      auto sfc_alb_dir_vis_k = subview_1dkc(d_sfc_alb_dir_vis);
+      auto sfc_alb_dir_nir_k = subview_1dkc(d_sfc_alb_dir_nir);
+      auto sfc_alb_dif_vis_k = subview_1dkc(d_sfc_alb_dif_vis);
+      auto sfc_alb_dif_nir_k = subview_1dkc(d_sfc_alb_dif_nir);
+      auto qc_k              = subview_2dkc(d_qc, m_nlay);
+      auto nc_k              = subview_2dkc(d_nc, m_nlay);
+      auto qi_k              = subview_2dkc(d_qi, m_nlay);
+      auto cldfrac_tot_k     = m_buffer.cldfrac_tot_k;
+      auto rel_k             = subview_2dkc(d_rel, m_nlay);
+      auto rei_k             = subview_2dkc(d_rei, m_nlay);
+      auto sw_flux_up_k      = subview_2dk(d_sw_flux_up, m_nlay+1);
+      auto sw_flux_dn_k      = subview_2dk(d_sw_flux_dn, m_nlay+1);
+      auto sw_flux_dn_dir_k  = subview_2dk(d_sw_flux_dn_dir, m_nlay+1);
+      auto lw_flux_up_k      = subview_2dk(d_lw_flux_up, m_nlay+1);
+      auto lw_flux_dn_k      = subview_2dk(d_lw_flux_dn, m_nlay+1);
+      auto sw_clnclrsky_flux_up_k      = subview_2dk(d_sw_clnclrsky_flux_up, m_nlay+1);
+      auto sw_clnclrsky_flux_dn_k      = subview_2dk(d_sw_clnclrsky_flux_dn, m_nlay+1);
+      auto sw_clnclrsky_flux_dn_dir_k  = subview_2dk(d_sw_clnclrsky_flux_dn_dir, m_nlay+1);
+      auto sw_clrsky_flux_up_k      = subview_2dk(d_sw_clrsky_flux_up, m_nlay+1);
+      auto sw_clrsky_flux_dn_k      = subview_2dk(d_sw_clrsky_flux_dn, m_nlay+1);
+      auto sw_clrsky_flux_dn_dir_k  = subview_2dk(d_sw_clrsky_flux_dn_dir, m_nlay+1);
+      auto sw_clnsky_flux_up_k      = subview_2dk(d_sw_clnsky_flux_up, m_nlay+1);
+      auto sw_clnsky_flux_dn_k      = subview_2dk(d_sw_clnsky_flux_dn, m_nlay+1);
+      auto sw_clnsky_flux_dn_dir_k  = subview_2dk(d_sw_clnsky_flux_dn_dir, m_nlay+1);
+      auto lw_clnclrsky_flux_up_k   = subview_2dk(d_lw_clnclrsky_flux_up, m_nlay+1);
+      auto lw_clnclrsky_flux_dn_k   = subview_2dk(d_lw_clnclrsky_flux_dn, m_nlay+1);
+      auto lw_clrsky_flux_up_k      = subview_2dk(d_lw_clrsky_flux_up, m_nlay+1);
+      auto lw_clrsky_flux_dn_k      = subview_2dk(d_lw_clrsky_flux_dn, m_nlay+1);
+      auto lw_clnsky_flux_up_k      = subview_2dk(d_lw_clnsky_flux_up, m_nlay+1);
+      auto lw_clnsky_flux_dn_k      = subview_2dk(d_lw_clnsky_flux_dn, m_nlay+1);
+      auto sw_bnd_flux_up_k  = m_buffer.sw_bnd_flux_up_k;
+      auto sw_bnd_flux_dn_k  = m_buffer.sw_bnd_flux_dn_k;
+      auto sw_bnd_flux_dir_k = m_buffer.sw_bnd_flux_dir_k;
+      auto sw_bnd_flux_dif_k = m_buffer.sw_bnd_flux_dif_k;
+      auto lw_bnd_flux_up_k  = m_buffer.lw_bnd_flux_up_k;
+      auto lw_bnd_flux_dn_k  = m_buffer.lw_bnd_flux_dn_k;
+      auto sfc_flux_dir_vis_k = subview_1dk(d_sfc_flux_dir_vis);
+      auto sfc_flux_dir_nir_k = subview_1dk(d_sfc_flux_dir_nir);
+      auto sfc_flux_dif_vis_k = subview_1dk(d_sfc_flux_dif_vis);
+      auto sfc_flux_dif_nir_k = subview_1dk(d_sfc_flux_dif_nir);
+      auto aero_tau_sw_k     = m_buffer.aero_tau_sw_k;
```

Comment:
These 4 are unfortunately needed because RRTMGP expects ordering (col,lay,bnd) but the FM keeps things in (col,bnd,lay) order.

Reply:
Is this not true for the cld_ ones below?

Reply:
It might be but those don't need to be initialized from field vars.